### PR TITLE
e2e_network:stop using deprecated framework.ExpectNotEqual

### PR DESCRIPTION
**Pull request by wlq1212**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
e2e_network:stop using deprecated framework.ExpectNotEqual

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: test/e2e/network/ingress.go [hunk 1]
```diff
@@ -786,7 +786,7 @@ var _ = common.SIGDescribe("Ingress API", func() {
 		ginkgo.By("deleting")
 
 		expectFinalizer := func(ing *networkingv1.Ingress, msg string) {
-			framework.ExpectNotEqual(ing.DeletionTimestamp, nil, fmt.Sprintf("expected deletionTimestamp, got nil on step: %q, ingress: %+v", msg, ing))
+			gomega.Expect(ing.DeletionTimestamp).ToNot(gomega.BeNil(), fmt.Sprintf("expected deletionTimestamp, got nil on step: %q, ingress: %+v", msg, ing))
```
Comment (carlory) [inline test/e2e/network/ingress.go:789]:
```suggestion
			gomega.Expect(ing.DeletionTimestamp).ToNot(gomega.BeNil(), "expected deletionTimestamp, got nil on step: %q, ingress: %+v", msg, ing)
```

File: test/e2e/network/service.go [hunk 2]
```diff
@@ -1401,7 +1401,7 @@ var _ = common.SIGDescribe("Services", func() {
 		gomega.Expect(nodePortCounts).To(gomega.Equal(2), "updated service should have two Ports but found %d Ports", nodePortCounts)
 
 		for _, port := range nodePortService.Spec.Ports {
-			framework.ExpectNotEqual(port.NodePort, 0, "NodePort service failed to allocate NodePort for Port %s", port.Name)
+			gomega.Expect(port.NodePort).ToNot(gomega.HaveValue(gomega.Equal(0)), "NodePort service failed to allocate NodePort for Port %s", port.Name)
```
Comment (carlory) [inline test/e2e/network/service.go:1404]: why not?

```suggestion
			gomega.Expect(port.NodePort).ToNot(gomega.BeZero(), "NodePort service failed to allocate NodePort for Port %s", port.Name)
```

Comment (wlq1212) [reply test/e2e/network/service.go]: resolved

File: test/e2e/network/loadbalancer.go [hunk 3]
```diff
@@ -552,6 +552,7 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 		framework.ExpectNoError(err, "Unable to get pod %s", acceptPod.Name)
 		gomega.Expect(acceptPod.Status.Phase).To(gomega.Equal(v1.PodRunning))
 		framework.ExpectNotEqual(acceptPod.Status.PodIP, "")
+		gomega.Expect(acceptPod.Status.PodIP).ToNot(gomega.Equal(""))
```
Comment (pohly) [inline test/e2e/network/loadbalancer.go:555]: The old `framework.ExpectNotEqual` hasn't been removed yet here.

**Review by pohly — CHANGES_REQUESTED** [review]
/lgtm cancel

File: test/e2e/network/loadbalancer.go [hunk 4]
```diff
@@ -581,7 +581,7 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
 		dropPod, err = cs.CoreV1().Pods(namespace).Get(ctx, dropPod.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err, "Unable to get pod %s", dropPod.Name)
 		gomega.Expect(acceptPod.Status.Phase).To(gomega.Equal(v1.PodRunning))
-		framework.ExpectNotEqual(acceptPod.Status.PodIP, "")
+		gomega.Expect(acceptPod.Status.PodIP).ToNot(gomega.Equal(""))
```
Comment (pohly) [inline test/e2e/network/loadbalancer.go:584]:
```suggestion
		gomega.Expect(acceptPod.Status.PodIP).ToNot(gomega.BeEmpty())
```

Comment (wlq1212) [reply test/e2e/network/loadbalancer.go]: thanks

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**